libsolidity/interface/CompilerStack.cpp

```diff
@@ -1013,15 +1016,34 @@ Json::Value const& CompilerStack::natspecDev(Contract const& _contract) const
 	return _contract.devDocumentation.init([&]{ return Natspec::devDocumentation(*_contract.contract); });
 }
 
-Json::Value CompilerStack::methodIdentifiers(string const& _contractName) const
+Json::Value CompilerStack::contractIdentifiers(string const& _contractName) const
 {
 	if (m_stackState < AnalysisPerformed)
 		solThrow(CompilerError, "Analysis was not successful.");
 
-	Json::Value methodIdentifiers(Json::objectValue);
+	Json::Value contractIdentifiers(Json::objectValue);
+	// Always have a methods object
+	contractIdentifiers["methods"] = Json::objectValue;
+
 	for (auto const& it: contractDefinition(_contractName).interfaceFunctions())
-		methodIdentifiers[it.second->externalSignature()] = it.first.hex();
-	return methodIdentifiers;
+		contractIdentifiers["methods"][it.second->externalSignature()] = it.first.hex();
+	for (ErrorDefinition const* error: contractDefinition(_contractName).interfaceErrors())
+	{
+		string signature = error->functionType(true)->externalSignature();
+		contractIdentifiers["errors"][signature] = toHex(toCompactBigEndian(selectorFromSignature32(signature), 4));
+	}
+
+	for (EventDefinition const* event: ranges::concat_view(
+		contractDefinition(_contractName).interfaceEvents(),
```

Comment:
Now this is not in sync with `interfaceErrors`, the two functions sound similar but they return totally different items.
Maybe either keep just a single function with a parameter or rename `interfaceEvents` to something like `definedInterfaceEvents`.

Reply:
In a PR to breaking, I changed the `interfaceEvents` function to be identical to `interfaceErrors`. Maybe we should just add that function as `interfaceEventsBreaking` or something instead of concatenating the vectors here again? Also note that most events will be in both vectors, I think that's confusing.